components/org.wso2.carbon.identity.oauth/src/test/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuerTest.java

```diff
@@ -290,8 +307,55 @@ public void testCreateJWTClaimSet(Object authzReqMessageContext,
 
     }
 
-    @Test
-    public void testSignJWTWithRSA() throws Exception {
+    @Test(dataProvider = "createJWTClaimSetDataProvider")
+    public void testSignJWTWithRSA(Object authzReqMessageContext,
+                                   Object tokenReqMessageContext,
+                                   String sub,
+                                   long expectedExpiry) throws Exception {
+
+        try {
+            OAuthAppDO appDO = spy(new OAuthAppDO());
+            mockGrantHandlers();
+            mockCustomClaimsCallbackHandler();
+            mockStatic(OAuth2Util.class);
+            when(OAuth2Util.getAppInformationByClientId(anyString())).thenReturn(appDO);
+            when(OAuth2Util.getThumbPrint(anyString(), anyInt())).thenReturn(THUMB_PRINT);
+
+            System.setProperty(CarbonBaseConstants.CARBON_HOME,
+                    Paths.get(System.getProperty("user.dir"), "src", "test", "resources").toString());
+            KeyStore wso2KeyStore = getKeyStoreFromFile("wso2carbon.jks", "wso2carbon",
+                    System.getProperty(CarbonBaseConstants.CARBON_HOME));
+            RSAPrivateKey rsaPrivateKey = (RSAPrivateKey) wso2KeyStore.getKey("wso2carbon", "wso2carbon".toCharArray());
+
+            when((OAuth2Util.getPrivateKey(anyString(), anyInt()))).thenReturn(rsaPrivateKey);
+            JWSSigner signer = new RSASSASigner(rsaPrivateKey);
+            when(OAuth2Util.createJWSSigner(any())).thenReturn(signer);
+            when(oAuthServerConfiguration.getSignatureAlgorithm()).thenReturn(SHA256_WITH_RSA);
+
+            mockStatic(IdentityTenantUtil.class);
+            when(IdentityTenantUtil.getTenantId(anyString())).thenReturn(-1234);
+
+            JWTTokenIssuer jwtTokenIssuer = PowerMockito.spy(new JWTTokenIssuer());
+            PowerMockito.doReturn(sub).when(jwtTokenIssuer, "getSubjectClaim", anyString(), anyString(), any());
+            JWTClaimsSet jwtClaimSet = jwtTokenIssuer.createJWTClaimSet(
+                    (OAuthAuthzReqMessageContext) authzReqMessageContext,
+                    (OAuthTokenReqMessageContext) tokenReqMessageContext,
+                    DUMMY_CLIENT_ID
+            );
+
+            String jwtToken = jwtTokenIssuer.signJWT(jwtClaimSet,
+                    (OAuthTokenReqMessageContext) tokenReqMessageContext,
+                    (OAuthAuthzReqMessageContext) authzReqMessageContext);
+            SignedJWT signedJWT = SignedJWT.parse(jwtToken);
+            assertNotNull(jwtToken);
+            assertNotNull(signedJWT.getHeader());
+            assertNotNull(signedJWT.getHeader().getType());
+            assertEquals(signedJWT.getHeader().getType().toString(), DEFAULT_TYP_HEADER_VALUE);
+
+        } catch (Exception ex) {
+            assertTrue(ex.getMessage() != null && ex.getMessage().contains("is not supported"),
```

Comment:
Shall we change this error message to something more appropriate?

Reply:
resolve from https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1750/commits/8fb2a219af2b096c568506aa29f88bd5ab984e42